### PR TITLE
Feat/strategy staking integration

### PR DESCRIPTION
**Pull request by Defcon023**

### CODE REVIEW

File: src/LiquidityReserves/remote/AssetReserves.sol [hunk 1]
```diff
@@ -124,4 +189,56 @@ contract AssetReserves is IAssetReserves{
         }
     }
 
+    function rebalance(address asset) external {
```
Comment (samtvlabs) [inline src/LiquidityReserves/remote/AssetReserves.sol:192]: Who should be allowed to call rebalance?

File: src/LiquidityReserves/remote/AssetReserves.sol [hunk 2]
```diff
@@ -124,4 +189,56 @@ contract AssetReserves is IAssetReserves{
...
+            uint256 withdrawalAmount = tokenInfo.targetAvailableLiquidity - balance;
+            if(withdrawalAmount > 0 && assetBalanceInStrategy > withdrawalAmount) {
+                IERC4626(tokenInfo.strategyVault).withdraw(withdrawalAmount, address(this), address(this));
+                emit AssetRebalanced();
+            }
+        }
+    }
+
+    function sendExcessYield(address asset, uint256 khalaniPercentageBps, address khalaniRewardReceiver) public onlyOwner {
+        require(khalaniPercentageBps <= 10000, "Percentage cannot exceed 10_000 basis points (100%).");
```
Comment (samtvlabs) [inline src/LiquidityReserves/remote/AssetReserves.sol:212]: Why have you chosen to use requires here over custom errors

File: test/AssetReserves.t.sol [hunk 3]
```diff
@@ -202,6 +211,123 @@ contract AssetReservesTest is Test{
...
+        vm.expectEmit(true,true,true,true,address(assetReserves));
+        emit LockOrBurn(user, tokens);
+        assetReserves.lockOrBurn(user, tokens);
+        vm.stopPrank();
+
+        assertEq(strategy.balanceOf(address (assetReserves)), amount);
+
+
+        //It should have withdrawn from strategy
+        uint256 strategyBalancePrev = strategy.balanceOf(address(assetReserves));
```
Comment (samtvlabs) [inline test/AssetReserves.t.sol:306]: Is this variable needed ?

File: test/AssetReserves.t.sol [hunk 4]
```diff
@@ -202,6 +211,123 @@ contract AssetReservesTest is Test{
...
+        //available limit is set to 50_000 USDC
+        assetReserves.rebalance(address(usdc));
+        uint256 prevRate = strategy.previewDeposit(1 ether);
+        //lets create some profit 5%
+        uint256 profitEarned = createProfit(500);
+
+        uint256 afterRate = strategy.previewDeposit(1 ether);
+        assertTrue(afterRate < prevRate);
+
+        Token[] memory tokes = new Token[](1);
```
Comment (samtvlabs) [inline test/AssetReserves.t.sol:318]: Do we need this variable ?

File: test/AssetReserves.t.sol (hunk 3 above)
Comment (Defcon023) [reply test/AssetReserves.t.sol:306]: Thanks , this is not needed it was copied from rebalance test code

File: test/AssetReserves.t.sol (hunk 4 above)
Comment (Defcon023) [reply test/AssetReserves.t.sol:318]: It is used in the vm.mockCall below to mock the call to nexus with the expected calldata

File: src/LiquidityReserves/remote/AssetReserves.sol (hunk 1 above)
Comment (Defcon023) [reply src/LiquidityReserves/remote/AssetReserves.sol:192]: Rebalance can be called by any address at any time, it just works under the constraint of targetAvailableLiquidity set for an asset, if anyone is calling rebalance they bear the gas cost of deposit / withdraw from the strategy 

Should we make this permissioned for our keepers?

**Review by samtvlabs — APPROVED** [review]

**Pull request closed without merging**